--- origsrc/shiboken-1.0.0/CMakeLists.txt	2011-03-02 10:46:13.000000000 -0600
+++ src/shiboken-1.0.0/CMakeLists.txt	2011-03-25 00:18:43.312000200 -0500
@@ -24,7 +24,7 @@ option(BUILD_TESTS "Build tests." TRUE)
 if(MSVC)
     set(CMAKE_CXX_FLAGS "/Zc:wchar_t- /GR /EHsc /DWIN32 /D_WINDOWS /D_SCL_SECURE_NO_WARNINGS")
 else()
-    if(CMAKE_HOST_UNIX)
+    if(CMAKE_HOST_UNIX AND NOT CYGWIN)
         set(CMAKE_CXX_FLAGS "${CMAKE_CXX_FLAGS} -Wall -fvisibility=hidden -Wno-strict-aliasing")
     endif()
     set(CMAKE_CXX_FLAGS_DEBUG "-g")
--- origsrc/shiboken-1.0.0/data/ShibokenConfig-spec.cmake.in	2011-03-02 10:46:13.000000000 -0600
+++ src/shiboken-1.0.0/data/ShibokenConfig-spec.cmake.in	2011-03-25 00:19:53.324922200 -0500
@@ -8,6 +8,8 @@
 SET(SHIBOKEN_INCLUDE_DIR "@CMAKE_INSTALL_PREFIX@/include/shiboken@shiboken_SUFFIX@")
 if(MSVC)
    SET(SHIBOKEN_LIBRARY "@LIB_INSTALL_DIR@/@CMAKE_SHARED_LIBRARY_PREFIX@shiboken@shiboken_SUFFIX@-@PYTHON_BASENAME@@LIBRARY_OUTPUT_SUFFIX@.lib")
+elseif(CYGWIN)
+    SET(SHIBOKEN_LIBRARY "@LIB_INSTALL_DIR@/@CMAKE_IMPORT_LIBRARY_PREFIX@shiboken@shiboken_SUFFIX@-@PYTHON_BASENAME@@LIBRARY_OUTPUT_SUFFIX@@CMAKE_IMPORT_LIBRARY_SUFFIX@")
 elseif(WIN32)
     SET(SHIBOKEN_LIBRARY "@CMAKE_INSTALL_PREFIX@/bin/@CMAKE_SHARED_LIBRARY_PREFIX@shiboken@shiboken_SUFFIX@-@PYTHON_BASENAME@@LIBRARY_OUTPUT_SUFFIX@@CMAKE_SHARED_LIBRARY_SUFFIX@")
 else()
--- origsrc/shiboken-1.0.0/libshiboken/shibokenmacros.h	2011-03-02 10:46:13.000000000 -0600
+++ src/shiboken-1.0.0/libshiboken/shibokenmacros.h	2011-03-25 00:18:16.043152600 -0500
@@ -24,7 +24,7 @@
 #define SHIBOKENMACROS_H
 
 // LIBSHIBOKEN_API macro is used for the public API symbols.
-#if defined _WIN32 || defined __CYGWIN__
+#if defined _WIN32
     #if LIBSHIBOKEN_EXPORTS
         #define LIBSHIBOKEN_API __declspec(dllexport)
     #else
